### PR TITLE
Some submit feedback

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Adds some submit feedback. Ref: https://github.com/yuvipanda/jupyterhub-configurator/issues/5

This is how it looks on success:
success

This is how it looks on failure:
error

Feedback very much welcome (my UI skills are almost non-existent :grimacing: )

### CODE REVIEW

**Comment by yuvipanda** [comment]
Thanks for opening this, @GeorgianaElena!

What do you think of making this more react-y? I think using getElementById and friends is generally frowned upon, and due to how re-rendering works might not do what you actually want it to do. 

Maybe we should use a `isSaving` variable with [useState](https://reactjs.org/docs/hooks-state.html), and render appropriate messages based on the state of that?

/cc @damianavila

**Comment by yuvipanda** [comment]
The UX looks good to me though.

**Comment by damianavila** [comment]
After pressing Submit, I was expecting from green alert to show you the update process was successful.
I noticed the button changes to "Saved", but this is a subtle change, IMHO.
Alternatively, you may change the button to green, but that is not in sync with the rest of the alerts you are showing...

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,21 +1,53 @@
...
   });
 }
 
+function onChange(setButtonVisibility, setAlertText, setAlertVariant, setAlertVisibility, setProgress) {
+  setAlertVisibility(true);
+  setButtonVisibility(true);
+  setProgress("Submit");
+  setAlertText("You have unsaved changes, click submit to save them.");
+  setAlertVariant("warning");
+}
```
Comment (damianavila) [inline frontend/src/App.js]: Having the parameters in a different order when compared with the body of the function rings something in my head... I know this is very minor but from a readability perspective, having the same ordering from left to right and from top to bottom could help the reader, IMIHO (this is also happening in the previous `updateConfig` function, I think).

Comment (GeorgianaElena) [reply frontend/src/App.js]: Great idea! Thanks @damianavila!

**Comment by GeorgianaElena** [comment]
In the last commits, I changed a bit how the submit feedback mechanism looks and behaves. The reason is that I noticed that any alert or button text modification made in the `onChange` handler sometimes makes the form change event to be lost and never arrive to `onSubmit` ([this thread](https://stackoverflow.com/questions/8013282/onsubmit-event-silently-killed-by-onchange) seems to explain what's going on).

The only solution I could think of, was to basically do nothing during `onChange` and have a [Toast](https://react-bootstrap.github.io/components/toasts/) that hides itself after a time period. All the changes to the submit button and this toast element, happen only `onSubmit` and page init. This is how it looks like now:

configurator-feedback

What do you think?

File: frontend/src/App.js [hunk 2]
```diff
@@ -51,16 +76,27 @@ const uiSchemaForSchema = (schema) => {
...
       widgets={customWidgets}
-      onError={console.log}
-    />
+      onError={console.log}>
+      <Toast onClose={() => setAlertVisibility(false)} show={alertShow} delay={2000} autohide>
+        <Toast.Body className={alertVariant}>{alertText}</Toast.Body>
+      </Toast>
+      <Button type="submit" variant="primary" disabled={isLoading}>
+        {isLoading ? 'Loading…' : 'Submit'}
+      </Button>
```
Comment (damianavila) [inline frontend/src/App.js]: What about putting the toast after the button?
From the gif you posted, having the toast above is causing the button to be "dancing" and users usually expect buttons to stay in their place.
WDYT?

Comment (GeorgianaElena) [reply frontend/src/App.js]: I experimented a bit with placing the toast in different places :thinking: 

* This is how it looks with the toast at the bottom:
toast-at-the-bottom

* This is how it looks with the toast in the right side:
toat-at-right

The submit button seems it's not well aligned with the rest of the form though.

What do you think?

Comment (damianavila) [reply frontend/src/App.js]: Both are OK, but if you go with the first option, the toast should be separated from the Submit button and if you go with the second option, you should try to fix that subtle change in the button height once the toast appears, IMHO.

**Comment by damianavila** [comment]
OK, from your last commit it seems you are putting the elements in a row to prevent that weird button displacement, I guess that should do the trick! Do you have any gif to see how that looks like now? Thanks!

**Comment by GeorgianaElena** [comment]
> OK, from your last commit it seems you are putting the elements in a row to prevent that weird button displacement, I guess that should do the trick! Do you have any gif to see how that looks like now? Thanks!

Yes, using the `Row` element fixed that weird left alignment :tada:   
This is how it looks like now:
right-toast-new

**Review by damianavila — APPROVED** [review]
I think this is probably ready for this iteration! 
You can always do some updates on the UI :wink: but that is not a blocker for this PR merge, IMHO.

**Comment by consideRatio** [comment]
Hey all! I saw @scottyhq providing feedback about the value of feedback from the submit button in a 2i2c issue. I then realized this was already a documented feature request and found this open PR about it! :D

@GeorgianaElena do you think this could be ready for merge as it is? If so then let's ping Yuvi for another review pass!

**Comment by GeorgianaElena** [comment]
> @GeorgianaElena do you think this could be ready for merge as it is? If so then let's ping Yuvi for another review pass!

I *think* so, but don't remember exactly. I know that at some point I ran into some issues with some form changes that got lost and didn't get saved, but from https://github.com/yuvipanda/jupyterhub-configurator/pull/7#issuecomment-822324184 maybe that's solved :eyes:

I can try to remember how I tested this and double-check if there's interest in it again ?

**Comment by yuvipanda** [comment]
\o/

**Pull request merged**